packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -66,54 +72,87 @@ function BlockBindingsPanelMenuContent( { fieldsList, attribute, binding } ) {
 		},
 		[ clientId, attribute ]
 	);
+
 	return (
-		<>
-			{ Object.entries( fieldsList ).map( ( [ name, fields ], i ) => (
-				<Fragment key={ name }>
-					<Menu.Group>
-						{ Object.keys( fieldsList ).length > 1 && (
-							<Menu.GroupLabel>
-								{ registeredSources[ name ].label }
-							</Menu.GroupLabel>
-						) }
-						{ Object.entries( fields )
-							.filter(
-								( [ , args ] ) => args?.type === attributeType
-							)
-							.map( ( [ key, args ] ) => (
-								<Menu.RadioItem
-									key={ key }
-									onChange={ () =>
-										updateBlockBindings( {
-											[ attribute ]: {
-												source: name,
-												args: { key },
-											},
-										} )
-									}
-									name={ attribute + '-binding' }
-									value={ key }
-									checked={ key === currentKey }
-								>
+		<Menu placement={ isMobile ? 'bottom-start' : 'left-start' }>
+			{ Object.entries( sources )
+				.filter( ( [ , source ] ) => {
```

Comment:
Can we rewrite this as follows?

```suggestion
			{ Object.values( sources )
				.filter( ( source ) => {
```

Reply:
Ah, nvm, looks like we need the key [further down](https://github.com/WordPress/gutenberg/pull/71542/files#diff-bff98e322c4269c50aa13e4b251beb9d53d2d0c3d4066406f735d2d49bed83a1R86).

Reply:
Although for the React element keys, `source.name` might also work?

Reply:
Ah, that's not a thing, apparently. Oh well, so let's stick with `sourceKey`, I guess 😅